source/glest_game/gui/gui.cpp

```diff
@@ -408,45 +409,58 @@ void Gui::hotKey(SDL_KeyboardEvent key) {
+	if(isKeyPressed(configKeys.getSDLKey("HotKeySelectIdleHarvesterUnit"),key) == true) {
 		selectInterestingUnit(iutIdleHarvester);
 	}
 	//else if(key == configKeys.getCharKey("HotKeySelectBuiltBuilding")) {
-	else if(isKeyPressed(configKeys.getSDLKey("HotKeySelectBuiltBuilding"),key) == true) {
+	if(isKeyPressed(configKeys.getSDLKey("HotKeySelectBuiltBuilding"),key) == true) {
 		selectInterestingUnit(iutBuiltBuilding);
 	}
 	//else if(key == configKeys.getCharKey("HotKeyDumpWorldToLog")) {
-	else if(isKeyPressed(configKeys.getSDLKey("HotKeyDumpWorldToLog"),key) == true) {
+	if(isKeyPressed(configKeys.getSDLKey("HotKeyDumpWorldToLog"),key) == true) {
 		 if(SystemFlags::getSystemSettingType(SystemFlags::debugSystem).enabled == true) {
 			 std::string worldLog = world->DumpWorldToLog();
 			SystemFlags::OutputDebug(SystemFlags::debugSystem,"In [%s::%s Line: %d] worldLog dumped to [%s]\n",__FILE__,__FUNCTION__,__LINE__,worldLog.c_str());
 		}
 	}
 	//else if(key == configKeys.getCharKey("HotKeyRotateUnitDuringPlacement")){
-	else if(isKeyPressed(configKeys.getSDLKey("HotKeyRotateUnitDuringPlacement"),key) == true) {
-	    // Here the user triggers a unit rotation while placing a unit
-	    if(isPlacingBuilding()) {
-	    	if(getBuilding()->getRotationAllowed()){
+	if(isKeyPressed(configKeys.getSDLKey("HotKeyRotateUnitDuringPlacement"),key) == true) {
+		// Here the user triggers a unit rotation while placing a unit
+		if(isPlacingBuilding()) {
+			if(getBuilding()->getRotationAllowed()){
 				++selectedBuildingFacing;
-	    	}
-	    }
+			}
+		}
 	}
 	//else if(key == configKeys.getCharKey("HotKeySelectDamagedUnit")) {
-	else if(isKeyPressed(configKeys.getSDLKey("HotKeySelectDamagedUnit"),key) == true) {
+	if(isKeyPressed(configKeys.getSDLKey("HotKeySelectDamagedUnit"),key) == true) {
 		selectInterestingUnit(iutDamaged);
 	}
 	//else if(key == configKeys.getCharKey("HotKeySelectStoreUnit")) {
-	else if(isKeyPressed(configKeys.getSDLKey("HotKeySelectStoreUnit"),key) == true) {
+	if(isKeyPressed(configKeys.getSDLKey("HotKeySelectStoreUnit"),key) == true) {
 		selectInterestingUnit(iutStore);
 	}
 	//else if(key == configKeys.getCharKey("HotKeySelectedUnitsAttack")) {
-	else if(isKeyPressed(configKeys.getSDLKey("HotKeySelectedUnitsAttack"),key) == true) {
+	if(isKeyPressed(configKeys.getSDLKey("HotKeySelectedUnitsAttack"),key) == true) {
 		clickCommonCommand(ccAttack);
 	}
 	//else if(key == configKeys.getCharKey("HotKeySelectedUnitsStop")) {
-	else if(isKeyPressed(configKeys.getSDLKey("HotKeySelectedUnitsStop"),key) == true) {
+	if(isKeyPressed(configKeys.getSDLKey("HotKeySelectedUnitsStop"),key) == true) {
 		clickCommonCommand(ccStop);
 	}
+
+	for (int i=0; i<10; i++) {
+		string name = "HotKeyBuild" + intToStr(i+1);
+		if(isKeyPressed(configKeys.getSDLKey(name.c_str()),key) == true) {
+			if(activeCommandType != NULL && activeCommandType->getClass() == ccBuild)  {
```

Comment:
you have hotkeys for the build menu! keep  in mind that every command when you select a unit is mapped, including the build command!

Reply:
In my testing, the grid of hotkeys only works when you click the worker's "build" menu. So you still have to move your mouse and click on the menu, to make the shortcuts work.

It would be very nice the keys grid worked at every menu level for every unit.
^ that would also solve the "I key double mapped" problem.

Reply:
That's exactly how it works (or it should) as I tried to describe. It's the else branch of the code you linked.
Press a worker, press E, the building menu opens, press A the building appears.

Reply:
Same for an attacking unit, press e, it's the attack command.

Reply:
It seems I found a way to reproduce the issue. The first time you click the worker it doesn't seem to work. If you click a unit and issue an attack command first with the hotkey it works and *then* the building menu works. Can you please confirm you can reproduce that?

Reply:
Found the issue, commonCommands was only updated for a non uniform selection and not for a uniform selection.

https://github.com/MegaGlest/megaglest-source/blob/ac0db40a7473122a80fcd2f78e4c52e40b13f5a4/source/glest_game/gui/gui.cpp#L1040

Reply:
> That's exactly how it works (or it should) as I tried to describe. It's the else branch of the code you linked. Press a worker, press E, the building menu opens, press A the building appears.

@Jammyjamjamman I tested with the recent changes; it works now.

Reply:
Works for me too :+1: